aws/resource_aws_sns_platform_application_test.go

```diff
@@ -0,0 +1,447 @@
+package aws
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	homedir "github.com/mitchellh/go-homedir"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sns"
+)
+
+/**
+ Before running this test, at least one of these ENV variables combinations must be set:
+
+ GCM_API_KEY - Google Cloud Messaging API Key
+
+ APNS_SANDBOX_CREDENTIAL_PATH - Apple Push Notification Sandbox Private Key file location
+ APNS_SANDBOX_PRINCIPAL_PATH - Apple Push Notification Sandbox Certificate file location
+**/
+
+type testAccAwsSnsPlatformApplicationPlatform struct {
+	Name           string
+	Credential     string
+	CredentialHash string
+	Principal      string
+	PrincipalHash  string
+}
+
+func testAccAwsSnsPlatformApplicationPlatformFromEnv() ([]*testAccAwsSnsPlatformApplicationPlatform, error) {
```

Comment:
Nitpick: This function could theoretically take `*testing.T` as argument and do the skipping which would save us ~3 lines in each test and more importantly allow us to `t.Fatal()` on actual errors.

Reply:
That is a fantastic idea!